pkg/controller/perconaxtradbcluster/perconaxtradbcluster_controller.go

```diff
@@ -214,7 +214,13 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	if cr.Spec.Platform != nil {
 		serverVersion.Platform = *cr.Spec.Platform
 	}
-
+	if len(cr.Spec.SSLSecretName) > 0 {
+		cr.Spec.PXC.SSLSecretName = cr.Spec.SSLSecretName
+		cr.Spec.ProxySQL.SSLSecretName = cr.Spec.SSLSecretName
+	} else {
+		cr.Spec.PXC.SSLSecretName = cr.Name + "-ssl"
+		cr.Spec.ProxySQL.SSLSecretName = cr.Name + "-ssl"
+	}
```

Comment:
I believe it worth to move this config setup to the `PerconaXtraDBCluster.CheckNSetDefaults()` https://github.com/percona/percona-xtradb-cluster-operator/blob/master/pkg/apis/pxc/v1alpha1/perconaxtradbcluster_types.go#L208.
It looks more logically there